packages/app-builder-lib/src/node-module-collector/nodeModulesCollector.ts

```diff
@@ -0,0 +1,120 @@
+import { hoist, type HoisterTree, type HoisterResult } from "./hoist"
+import * as path from "path"
+import * as fs from "fs"
+import { NodeModuleInfo, DependencyTree, DependencyGraph } from "./types"
+import { log } from "builder-util"
+
+export abstract class NodeModulesCollector {
+  private nodeModules: NodeModuleInfo[]
+  protected rootDir: string
+  protected dependencyPathMap: Map<string, string>
+
+  constructor(rootDir: string) {
```

Comment:
Pro tip, you can use `constructor(private readonly rootDir: string)` to instantiate vars with the class.

```
export abstract class NodeModulesCollector {
  private nodeModules: NodeModuleInfo[] = []
  protected dependencyPathMap = new Map<string, string>()

  constructor(private readonly rootDir: string) {}
```